drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/projects_importer.rb

```diff
@@ -0,0 +1,198 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::AcHmis::Importers
+  class ProjectsImporter
+    include NotifierConfig
+
+    AbortImportException = Class.new(StandardError)
+
+    attr_accessor :attempt
+    attr_accessor :data_source
+    attr_accessor :dir
+
+    def initialize(dir:, key:, etag:)
+      self.attempt = ProjectsImportAttempt.where(etag: etag, key: key).first_or_initialize
+      self.data_source = HmisExternalApis::AcHmis.data_source
+      self.dir = dir
+    end
+
+    def run!
+      start
+      sanity_check
+      ProjectsImportAttempt.transaction do
+        upsert_funders
+        upsert_orgs
+        upsert_projects
+        upsert_walkins
+      end
+      analyze
+      finish
+    rescue AbortImportException => e
+      @notifier.ping('Failure in project importer', { exception: e })
+      Rails.logger.fatal e.message
+    end
+
+    def start
+      setup_notifier('HMIS Projects')
+      Rails.logger.info "Starting #{attempt.key}"
+      attempt.attempted_at = Time.current
+      attempt.status = ProjectsImportAttempt::STARTED
+      attempt.save!
+    end
+
+    def sanity_check
```

Comment:
If the zip files have already been expanded at this point, it's probably worth checking the headers against a known set of headers and throwing an error if they don't match.  Maybe also that we received all the expected files?